mavros/src/plugins/setpoint_position.cpp

```diff
@@ -146,7 +146,7 @@ class SetpointPositionPlugin : public plugin::PluginBase,
 
 		auto q = [&] () {
 				if (mav_frame == MAV_FRAME::BODY_NED || mav_frame == MAV_FRAME::BODY_OFFSET_NED) {
-					return ftf::transform_orientation_baselink_aircraft(Eigen::Quaterniond(tr.rotation()));
+					return ftf::quaternion_from_rpy(M_PI, 0.0, 0.0)*Eigen::Quaterniond(tr.rotation());
```

Comment:
This is exactly the same rotation as the one in https://github.com/mavlink/mavros/blob/master/mavros/include/mavros/frame_tf.h#L185. What exactly are you solving here?